File: hw/cañarte/u1/HW09FromCDtoPython/ec_edu_espe_farmsimulator_model/Chicken.py
from datetime import date 


class Chicken:
    def __init__(self, id, name, color, age, molting, born_on_date):
        self.id = id
        self.name = name
        self.color = color
        self.age = age
        self.molting = molting
        self.born_on_date = born_on_date

    def __str__(self):
        return f"Chicken(id={self.id}, name={self.name}, color={self.color}, age={self.age}, molting={self.molting}, bornOnDate={self.born_on_date})"
    

    def get_id(self):
        return self.__id
    def get_name (self):
        return self.__name
    def get_color (self):
        return self.__color
    def get_age (self):
        return self.__age
    def get_molting (self):
        return self.__molting
    def get_born_on_date (self):
        return self.__born_on_date

    
    
    

File: hw/Rodriguez/u1/hw_10/ChikenFarmSimulator/src/ec/edu/espe/farmsimulator/model/Chicken.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.farmsimulator.model;



import java.text.SimpleDateFormat;
import java.util.Date;
import utils.FileManager;


/**
 *
 * @author David Rodriguez,THEJAVABANDITS,DCCO-ESPE
 */
public class Chicken {
    private int id;
    private String name;
    private String color;
    private int age;
    private boolean molting;
    private Date bornOnDate;

    public Chicken(int id, String name, String color, boolean molting, int age, Date bornOnDate) {
        this.id = id;
        this.name = name;
        this.color = color;
        this.age = age;
        this.molting = molting;
        this.bornOnDate = bornOnDate;
    }

    @Override
    public String toString() {
        SimpleDateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
        String formattedDate = dateFormat.format(bornOnDate); 
        return id + "," + name + "," + color + "," + age + "," + molting + "," + formattedDate;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public boolean isMolting() {
        return molting;
    }

    public void setMolting(boolean molting) {
        this.molting = molting;
    }

    public Date getBornOnDate() {
        return bornOnDate;
    }

    public void setBornOnDate(Date bornOnDate) {
        this.bornOnDate = bornOnDate;
    }
}